### PR TITLE
all method to get all messages from local message cache queue

### PR DESCRIPTION
**Pull request by michaelclapham**
all method to allow getting multiple valid messages from the local message cache queue.

### CODE REVIEW

File: src/logic/LocalMessageCache.tsx [hunk 1]
```diff
@@ -132,6 +137,23 @@ class LocalMessageCache {
...
     }
 
+    static all(
+        slotName: SlotName,
+        appboyInstance: typeof appboy,
+        errorHandler: ErrorHandler,
+    ): MessageWithId[] {
+        const queue = getQueue(slotName, errorHandler);
+
+        if (queue) {
```
Comment (tjmw) [inline src/logic/LocalMessageCache.tsx:147]: No harm, but looking at the way `peek` uses `getQueue` and the return type of `getQueue` I'm not sure this condition is necessary. It looks like we'll always get an array of type `CachedMessage[]` back so we can probably always just return `queue.map(...)` here. What do you reckon?

**Review by tjmw — COMMENTED** [review]
Nice! This is looking good. I left a couple of minor questions/suggestions, see what you think. Happy to chat over if anything isn't clear.

File: src/logic/LocalMessageCache.test.ts [hunk 2]
```diff
@@ -50,6 +50,140 @@ const buildExpiredMessage = (message: Message, id: string): CachedMessage => ({
...
+
+        it('returns empty list if the queue is empty', () => {
+            setQueue('EndOfArticle', []);
+
+            const messages = LocalMessageCache.all('EndOfArticle', appboy, noopErrorHandler);
+
+            expect(messages.length).toEqual(0);
+        });
+
+        it('returns the first unexpired message', () => {
```
[the remark this answers is not in the document]
Comment (michaelclapham) [reply src/logic/LocalMessageCache.test.ts]: Fixed in https://github.com/guardian/braze-components/pull/151/commits/5527dd89c3da9a5fddfc48473e1892678d6132b2

File: src/logic/LocalMessageCache.tsx (hunk 1 above)
Comment (michaelclapham) [reply src/logic/LocalMessageCache.tsx]: Fixed in https://github.com/guardian/braze-components/pull/151/commits/ec8a83027ff6992bcef55ef2b9f874c5a3b02e26

**Review by tjmw — APPROVED** [review]
💯

**Pull request merged**